source/Variant.cpp

```diff
@@ -0,0 +1,104 @@
+/* Variant.cpp
+Copyright (c) 2022 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Variant.h"
+
+#include "DataNode.h"
+#include "GameData.h"
+#include "Ship.h"
+
+#include <algorithm>
+
+using namespace std;
+
+
+
+Variant::Variant(const DataNode &node)
+{
+	Load(node);
+}
+
+
+
+void Variant::Load(const DataNode &node)
+{
+	if(node.Token(0) == "variant" && node.Size() >= 2)
+		weight = node.Value(1);
+	else if(node.Token(0) == "add" && node.Size() >= 3)
+		weight = node.Value(2);
```

Comment:
What will happen if the user provides a negative / 0 weight here? Game exits with an exception?